src/silo/query_engine/query_engine.cpp

```diff
@@ -74,86 +75,20 @@ response::QueryResult QueryEngine::executeQuery(const std::string& query) const
    for (unsigned i = 0; i < database.partitions.size(); ++i) {
       SPDLOG_DEBUG("Simplified query for partition {}: {}", i, compiled_queries[i]);
    }
-   LOG_PERFORMANCE("Execution (filter): {} microseconds", std::to_string(query_result.filter_time));
+   LOG_PERFORMANCE("Execution (filter): {} microseconds", std::to_string(filter_time));
+
+   auto action = json["action"].get<std::unique_ptr<silo::query_engine::actions::Action>>();
```

Comment:
If this fails, we already did the potentially expensive filtering of the DB -> Let's move this up?

I'd even propose to create another struct `SiloQuery { filter_expression, action }` and simply call `json.get<SiloQuery>()`.

Reply:
In a later PR, when we have the actions implemented as well?

Reply:
Or maybe simply in a separate commit? I don't see a reason why we need to wait for the other actions?